### ISSUE TITLE
WeightDecay argument in ADAMW optimiser

### ISSUE THREAD

**Issue by PastelBelem8** [FluxML/Flux.jl#507]
https://github.com/FluxML/Flux.jl/blob/a32c8a2e60870d49475719a36c74afed305e370a/src/optimise/optimisers.jl#L231 is throwing `UndefVarError: wd not defined`

I think `wd` should be replaced by the value of parameter `decay`.

So it would be
 ```julia
ADAMW(η = 0.001, β = (0.9, 0.999), decay = 0) =
  Optimiser(ADAM(η, β), WeightDecay(decay))
```

**Comment by dhairyagandhi96** [comment]
Yes, thanks for that, it also looks like we missed it in the tests, else it would've been caught. Would you be up to open a PR for it?

**Comment by dhairyagandhi96** [comment]
Yeah sure, do also take a look at the existing testing scheme to get an idea

**Comment by dhairyagandhi96** [comment]
Would be happy to close this one in case you had something different in mind. Conversely, feel free to add to it.

**Comment by PastelBelem8** [comment]
I can do it @dhairyagandhi96. I'll look at until the end of this week (: 
I've been working on another project that drove my attention for a few days.

**Comment by dhairyagandhi96** [comment]
Fixed on master + current release 0.7.1

Thanks for this!
